src/main/groovy/org/gradle/android/workarounds/room/operations/FileSchemaOperations.groovy

```diff
@@ -0,0 +1,22 @@
+package org.gradle.android.workarounds.room.operations
+
+import org.gradle.api.file.Directory
+import org.gradle.api.internal.file.FileOperations
+import org.gradle.api.provider.Provider
+
+class FileSchemaOperations {
+    FileOperations fileOperations
+
+    FileSchemaOperations(FileOperations fileOperations) {
+        this.fileOperations = fileOperations
+    }
+
+    void sync(Provider<Directory> origin, Provider<Directory> destination) {
+        if (origin.isPresent()) {
```

Comment:
In the old code on line 319 this check was not performed. Could that lead to silent failures or is it not important?

Reply:
good question, the original check was placed in the `copyExistingSchemasToTaskSpecificTmpDir` method:
https://github.com/gradle/android-cache-fix-gradle-plugin/blob/main/src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy#L292
This method is called by `copyExistingSchemasToTaskSpecificTmpDirForKapt` and `copyExistingSchemasToTaskSpecificDirForJavaCompile`. 
I extracted the check in the `FileOperations` logic.